mail/inc/class.mail_compose.inc.php

```diff
@@ -615,6 +622,24 @@ function compose(array $_content=null,$msg=null, $_focusElement='to',$suppressSi
 			}
 
 		}
+		//hook mail_compose_index
```

Comment:
I believe you can just use the existing [etemplate2_before_exec](https://github.com/EGroupware/egroupware/blob/master/api/src/Etemplate.php#L105) hook, which also allows to add actions to the mail index page. As an example check the Collabora App: https://github.com/EGroupware/collabora/blob/master/src/Ui.php#L57

Ralf